### ISSUE TITLE
wrong user path when loading file

### ISSUE THREAD

**Issue by mmahalwy** [jtangelder/sass-loader#126]
See: https://github.com/justin808/bootstrap-sass-loader/issues/22

So it finds and logs the file, all is good. And now it cannot load it for some reason and pointing to the wrong please for the error. All in all, it cannot import the bootstrap-config file...

here is the log:
```
[boostrap-sass-loader]: styleLoader: /Users/mmahalwy/Desktop/Code/quran.com/frontend/node_modules/extract-text-webpack-plugin/loader.js?{"omit":1,"extract":true,"remove":true}!style-loader!css-loader!autoprefixer!sass?outputStyle=expanded
[boostrap-sass-loader]: bootstrap-sass location: /Users/mmahalwy/Desktop/Code/quran.com/frontend/node_modules/bootstrap-sass/assets
[boostrap-sass-loader]: Setting: $icon-font-path: 'node_modules/bootstrap-sass/assets/fonts/bootstrap/';
[boostrap-sass-loader]: fileName for preBootstrapCustomizations: /Users/mmahalwy/Desktop/Code/quran.com/frontend/src/styles/_bootstrap-config.scss
[boostrap-sass-loader]: fileName for mainSass: /Users/mmahalwy/Desktop/Code/quran.com/frontend/src/styles/_main.scss
Hash: 0865bbc60349c612e85a
Version: webpack 1.10.3
Time: 15497ms
      Asset       Size  Chunks             Chunk Names
    main.js       2 MB       0  [emitted]  main
main.js.map    2.33 MB       0  [emitted]  main
 index.html  211 bytes          [emitted]
   [0] multi main 64 bytes {0} [built]
    + 491 hidden modules

ERROR in ./~/extract-text-webpack-plugin/loader.js?{"omit":1,"extract":true,"remove":true}!./~/style-loader!./~/css-loader!./~/autoprefixer-loader!./~/sass-loader?outputStyle=expanded!./~/bootstrap-sass-loader/bootstrap-sass-styles.loader.js!./bootstrap-sass.config.js
Module build failed: ModuleBuildError: Module build failed:
var ExtractTextPlugin = require("extract-text-webpack-plugin");
       ^
      File to import not found or unreadable: Users/mmahalwy/Desktop/Code/quran.com/frontend/src/styles/bootstrap-config.scss
      in /Users/mmahalwy/Desktop/Code/quran.com/frontend/bootstrap-sass.config.js (line 1, column 9)
    at DependenciesBlock.onModuleBuildFailed (/Users/mmahalwy/Desktop/Code/quran.com/frontend/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:315:19)
    at nextLoader (/Users/mmahalwy/Desktop/Code/quran.com/frontend/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:270:31)
    at /Users/mmahalwy/Desktop/Code/quran.com/frontend/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:292:15
    at context.callback (/Users/mmahalwy/Desktop/Code/quran.com/frontend/node_modules/webpack/node_modules/webpack-core/lib/NormalModuleMixin.js:148:14)
    at Object.onRender (/Users/mmahalwy/Desktop/Code/quran.com/frontend/node_modules/sass-loader/index.js:160:13)
    at options.error (/Users/mmahalwy/Desktop/Code/quran.com/frontend/node_modules/node-sass/lib/index.js:279:32)
 @ ./~/bootstrap-sass-loader/index.loader.js!./bootstrap-sass.config.js 1:0-408
Child extract-text-webpack-plugin:
        + 1 hidden modules

    ERROR in ./~/css-loader!./~/autoprefixer-loader!./~/sass-loader?outputStyle=expanded!./~/bootstrap-sass-loader/bootstrap-sass-styles.loader.js!./bootstrap-sass.config.js
    Module build failed:
    var ExtractTextPlugin = require("extract-text-webpack-plugin");
           ^
          File to import not found or unreadable: Users/mmahalwy/Desktop/Code/quran.com/frontend/src/styles/bootstrap-config.scss
          in /Users/mmahalwy/Desktop/Code/quran.com/frontend/bootstrap-sass.config.js (line 1, column 9)
```

I should mention that my files are named correctly: 
![image](https://cloud.githubusercontent.com/assets/3749095/8844319/a4784c20-30c8-11e5-8bda-e081598b8bda.png)

For some reason, it's looking for the non-partial file. When I remove that, then it errors saying it cannot find variables.scss from bootstrap. There must be something wrong with the import system? 

*update*
Notice: `unreadable: Users/mmahalwy/` it should be `unreadable: /Users/mmahalwy/`....

**Comment by jorrit** [comment]
I am sorry to hear that you spent so much time on this and thanks for the bug report. Is your code available publicly so I can get the situation you have? Or can you describe the steps to reproduce this problem?

**Comment by jhnns** [comment]
Yes, a public example would be helpful. Besides that: Which version of sass-loader do you use? We've fixed some bugs with 1.0.3 concerning the import system.

**Comment by jorrit** [comment]
In the referenced issue is mentioned that 1.0.3 actually introduced the bug... so this is a regression.

**Comment by jhnns** [comment]
Ok, thx. Then we need a test to reproduce it, so we won't introduce the regression again.

**Comment by mmahalwy** [comment]
@jhnns I am using 1.0.3 and when downgrading to 1.0.2 it was fixed.

code is public here: https://github.com/quran/quran.com-frontend

**Comment by jhnns** [comment]
A minimal test setup would be more helpful. After 5 minutes I'm still cloning the repo... ^^

**Comment by ALF-er** [comment]
Hi guys. Have similar issue on Windows 10. Small example: https://github.com/ALF-er/wp_test simple import of bootstrap.css from node_modules. But receive error:
```
ERROR in ./~/css-loader!./~/sass-loader?includePaths[]=C:/dev/projects/wp_test/node_modules!./src/app/entrypoints/main.scss
Module not found: Error: Cannot resolve 'file' or 'directory' ./C:\dev\projects\wp_test\node_modules\bootstrap\dist\css\bootstrap.css in C:\dev\projects\wp_test\src\app\entrypoints
 @ ./~/css-loader!./~/sass-loader?includePaths[]=C:/dev/projects/wp_test/node_modules!./src/app/entrypoints/main.scss 2:10-146
```

**Comment by mmahalwy** [comment]
Yeah, it's because the fonts are huge :( 

But it shouldn't take longer than 5mins! :)

**Comment by jorrit** [comment]
`urlToRequest` is called with an absolute path, like `/Users/mmahalwy/...`. `loader-utils`' `urlToRequest` prepends `./`. Because of change [1] `.//Users/...` is translated to `Users/`.

If bool `true` or string `~` are passed to urlToRequest instead of `opt.root`, it works. I don't know what `opt.root` is supposed to do.

@jhnns: maybe change the code to `return utils.urlToRequest(url, opt.root || '~');` ?


[1] https://github.com/jtangelder/sass-loader/commit/94d0bc857aa0ef14790926183e88602bab7c2506#diff-168726dbe96b3ce427e7fedce31bb0bcR234

**Comment by jorrit** [comment]
This was fixed in justin808/bootstrap-sass-loader#25.

**Comment by mmahalwy** [comment]
@jorrit closing!
